test_util/test_runner.py

```diff
@@ -117,31 +126,43 @@ def integration_test(
         aws_secret_access_key: needed for REXRAY tests
         region: string indicating AWS region in which cluster is running
     """
-    log.info('Transfering integration_test.py')
+
+    log.info('Writing integration test Dockerfile')
+    dns_search = 'true' if test_dns_search else 'false'
+    temp_dir = tempfile.mkdtemp()
+    with open(join(temp_dir, 'Dockerfile'), 'w') as tempf:
+        tempf.write(TEST_DOCKERFILE.format(
+                dcos_dns=dcos_dns,
+                master_hosts=','.join(master_list),
+                slave_hosts=','.join(agent_list),
+                public_slave_hosts=','.join(public_agent_list),
+                registry_host='127.0.0.1',
+                dns_search=dns_search,
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                region=region))
+
+    # image uses ONBUILD so put the appropriate resources in the temp/build folder
+    log.info('Building and exporting integration test docker image')
+    pytest_docker = pkg_filename('docker/py.test/Dockerfile')
     test_script = pkg_filename('integration_test.py')
+    check_call(['docker', 'build', '-t', 'py.test:noop', '-'], stdin=open(pytest_docker))
+    shutil.copy(test_script, join(temp_dir, 'integration_test.py'))
+    check_call(['docker', 'build', '-t', 'py.test:run', '.'], cwd=temp_dir)
```

Comment:
The point of leaving it in two build steps is that this docker image is used test_util/run-all so and I did not want to disturb the flow there. Would you prefer to see them both use this approach? I would like if they could keep sharing the same Dockerfile